tests/system/datastore/test_v3io.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import shlex
+import subprocess
+
+import pytest
+
+import mlrun.datastore
+from tests.system.base import TestMLRunSystem
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.enterprise
+class TestV3ioDataStore(TestMLRunSystem):
+    @staticmethod
+    def _skip_set_environment():
+        return True
+
+    def test_v3io_large_object(self):
+        import random
+
+        with open("tempfile_1", "wb") as f:
+            file_size = 20 * 1024 * 1024  # 20MB
+            f.truncate(file_size)
+            for i in range(min(100, file_size)):
+                offset = random.randint(0, file_size - 1)
+                f.seek(offset)
+                f.write(bytearray([i]))
+        v3io_path = "v3io:///bigdata/test_v3io_large_object"
+        data_item = mlrun.datastore.store_manager.object(v3io_path)
+
+        data_item.upload("tempfile_1")
+        data_item.download("tempfile_2")
+
+        command = shlex.split("cmp tempfile_1 tempfile_2")
+        process = subprocess.Popen(command, stdout=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        assert (
+            process.returncode == 0
+        ), f"stdout = {stdout}, stderr={stderr}, returncode={process.returncode}"
+        data_item.delete()
+        os.remove("tempfile_1")
+        os.remove("tempfile_2")
```

Comment:
These files will not be deleted in case of an error, or when the assert fails. Better do this in a finally, or better yet, use [tempfile](https://docs.python.org/3/library/tempfile.html) as mentioned earlier.

Reply:
i'm not sure i can replace tempfile_2 with tempfile functionality: I assume tempfile works by opening a file-handle on the file and then deleting it. In that case, the file downloaded from iguazio will be a new file (with the same bind) and as a result will not be deleted when the test complete.
I changed the logic to assert at the end to reduce the chance of leftovers.

Reply:
I did link to https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp, which doesn't require using a handle.

Reply:
this one does not have guarantee of automatic cleanup so I'm not sure what is the value in using it..

Reply:
Alrighty, then [TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory)?

Reply:
done